### ISSUE TITLE
Removing a path could lead to leftover flows.

### ISSUE THREAD

**Issue by Alopalao** [kytos-ng/mef_eline#476]
While exploring from issue [#45](https://github.com/kytos-ng/kytos/issues/479):
> i. Allocate and deallocate many individual vlans on an interface

#### The issue in `failover_path` (`current_path` is similar):
- At the start of a `failover_path` installation, every needed VLAN is [removed](https://github.com/kytos-ng/mef_eline/blob/2e45177240ec9e9a94204494f99f90d7e3e94cf7/models/evc.py#L919) from the interfaces.
- The installation could [fail](https://github.com/kytos-ng/mef_eline/blob/2e45177240ec9e9a94204494f99f90d7e3e94cf7/models/evc.py#L938) in the middle of sending flows.
E.g:
```
failover_path = {
"00:00:00:00:00:00:00:01:1",
"00:00:00:00:00:00:00:01:4", "00:00:00:00:00:00:00:03:3",    # Using VLAN 87, correctly installed
"00:00:00:00:00:00:00:03:2", "00:00:00:00:00:00:00:02:3",    # Using VLAN 87, failed installation
"00:00:00:00:00:00:00:02:1"
}
```
- In this case the error was `Service Unavailable` which also is the error when trying to delete the flows from the paths.
- All the VLANs used by the path are freed, [code](https://github.com/kytos-ng/mef_eline/blob/2e45177240ec9e9a94204494f99f90d7e3e94cf7/models/evc.py#L795).
- Notice that we have 4 switches which VLANs do not match their installed flows:
```
"00:00:00:00:00:00:00:01:4"     # Freed VLAN 87, Flow was not removed
"00:00:00:00:00:00:00:03:3"     # Freed VLAN 87, Flow was not removed
"00:00:00:00:00:00:00:03:2"     # Freed VLAN 87, Flow was not installed
"00:00:00:00:00:00:00:02:3",    # Freed VLAN 87, Flow was not installed
```
- The next EVC will try its luck with the same VLAN 87 if it successfully installs its `failover_path`. It will install a flow in the same switch, port and VLAN.

- TLDR, when it comes to paths, we create/delete flows by switch but add/delete VLANs by paths. Deleting VLANs by confirmed deleted switch flow could solve this issue :thinking:.

#### Things to keep in mind:
- This issue was discovered when sending 2245 requests to create EVCs in ~7 threads~(turned out I created 2000 threads, thus the `Service Unavailable` error. Still this error is real.). This issue can be replicated by modifying `flow_manager` to detect a flow (a second or later addition flow) that is from a first EVC so a leftover installed flow is untouched. Then allowing every flow from a second EVC
- The test reproduced `Service Unavailable` errors from both `flow_manager` and `pathfinder`. But, only error from `flow_manager` is important for this issue.
- This issue does not seem critical. In my tests it only produced duplicated flows that would trigger constant consistency checks about a missing flow in a certain switch (related to the fail to remove path).
- This error can occur in [remove_path_flows](https://github.com/kytos-ng/mef_eline/blob/2e45177240ec9e9a94204494f99f90d7e3e94cf7/models/evc.py#L743-L799) and [remove_current_flows](https://github.com/kytos-ng/mef_eline/blob/2e45177240ec9e9a94204494f99f90d7e3e94cf7/models/evc.py#L708-L741) although it was much more likely in the first in `remove_path_flows`. In practice I have not noticed an issue in practice with `remove_current_flows` but the steps for dealing with errors is the same.

### Possible solutions:
- Closing issue #495 if ensures deletion of leftover flows at some point.
- Another solution would be to modify VLANs availability for confirmed flows installation and deletion only. Currently kytos modifies VLANs by path when these are not installed/deleted entirely.

#### Update
- Duplicated flows is not longer a problem from this scenario because [issue](https://github.com/kytos-ng/of_core/issues/136) from `of_core` was solved.

**Comment by Alopalao** [comment]
Created a not ideal [solution](https://github.com/kytos-ng/mef_eline/commit/22ec231f999026b788fd473e3b01c0a966bf725e). I solves duplicated flows for this case only.

**Comment by viniarck** [comment]
@Alopalao good finding, let's keep this in the backlog, since the service unavailable was due to a very high rate of requests that we don't expect in prod atm. As consistency check gets implemented it can also aid this case, and this one is already in progress, so let's avoid any other adjacent implementation in the meantime here. In the future, we can reassess if we'll augment the error handling part here and consider other adjacent efforts too like replacing requests or not.

**Comment by viniarck** [comment]
@Alopalao can this be closed or was there something remaining to still be mapped and addressed?

**Comment by Alopalao** [comment]
This issue can still be present. The solution for this issue is ensure flow installation without errors or allocate/deallocate VLANS by confirmed flow installation/deletion only.

**Comment by viniarck** [comment]
> This issue can still be present. The solution for this issue is ensure flow installation without errors or allocate/deallocate VLANS by confirmed flow installation/deletion only.

Right. When you have the chance, let's clarify and map the remaining potential errors that you mean here. Are they the ones related to this issue here https://github.com/kytos-ng/mef_eline/issues/495? If there's anything else let's document in the issue body just so we get back to it later when gets prioritized

**Comment by Alopalao** [comment]
Added `Possible solutions` so it is documented when to close this issue.

**Comment by Alopalao** [comment]
With now a single request being [sent per path](https://github.com/kytos-ng/mef_eline/issues/514), this issue does not longer exist. This issue can be closed.

**Comment by Alopalao** [comment]
Now thinking more about this, if the single request does gets the error `Service Unavailable` all their VLANS are going to still be freed which results in the same problem
